### DISCUSSION TITLE
Skip discover phase, jump to request

### DISCUSSION THREAD

**Discussion by rmrfchik** [rsmarples/dhcpcd#50] [Q&A]
I have a may be ill designed hardware which I cannot change. Ths hw works like wifi access point. Seems like DHCP works bad and hw don't answer to DISCOVER always and dhcpcd loops in DISCOVER messages. In rare cases hw answer to DISCOVER and subsequent REQUEST works fine. 
Android successfully connects to this AP in all cases.
Windows connects fine only if it has previous ip address (in this case Windows skips discover straight to request).

Is there way to skip DISCOVER in dhcpcd? Should it be configuration option?

**Comment by rsmarples** [comment]
Sorry for the late reply, but I have thought about this some.

The answer is "no". If dhcpcd has an active lease saved, it will automatically enter the REQUEST phase to start with.
If there is no answer within the short timeout period or there is no active lease then it will enter the DISCOVER phase.
We cannot REQUEST an address we don't know.

There are two options you have - working out which DHCP option the upstream DHCP server doesn't like (this is the most likely solution) or INFORMing a static IP address within the upstream DHCP servers address range.
